boundary_extraction.py

```diff
@@ -0,0 +1,84 @@
+import os
+import cv2
+import logging
+import argparse
+from pathlib import Path
+
+from tqdm import tqdm
+from tools.models import LungSegmentation
+from tools.utils import MorphologicalTransformations, get_file_list
+
+os.makedirs('logs', exist_ok=True)
+logging.basicConfig(
+    format='%(asctime)s - %(levelname)s - %(message)s',
+    datefmt='%d.%m.%Y %I:%M:%S',
+    filename='logs/{:s}.log'.format(Path(__file__).stem),
+    filemode='w',
+    level=logging.INFO,
+)
+logger = logging.getLogger(__name__)
+
+
+def boundary_extraction(
+        img_dir: str,
+        model_dir: str,
+        output_dir: str,
+        thresh_method: str = 'otsu',
+        thresh_val: float = 0.5,
+) -> None:
+
+    logger.info(f'Settings..................:')
+    logger.info(f'Image directory...........: {img_dir}')
+    logger.info(f'Output directory..........: {output_dir}')
+    logger.info(f'Threshold method..........: {thresh_method.capitalize()}')
+    logger.info(f'Threshold value...........: {thresh_val}')
+
+    model = LungSegmentation(
+        model_dir=model_dir,
+        threshold=0.50,
+        device='auto',
+        raw_output=True,
+    )
+
+    img_paths = get_file_list(
+        src_dirs=img_dir,
+        ext_list=[
+            '.png',
+            '.jpg',
+            '.jpeg',
+            '.bmp',
+        ]
+    )
+
+    for img_path in tqdm(img_paths, desc='Boundary extraction', unit=' images'):
+        # img_path = img_dir + '/' + img_path
+        mask = model(img_path)
+        cv2.imwrite(f'dataset/output/masks/mask_{img_path}.png', mask)
+        morph = MorphologicalTransformations(
```

Comment:
@sriva-e 

Revise **MorphologicalTransformations** because you should not create its object every iteration of the _for loop_. You have to create the **morph** before the loop and then use its methods in the loop. In that regard, you have to pass **thresh_method** and **thresh_val** to the class constructor, while the binary method should process **mask**.